### ISSUE TITLE
feat: attention — a key padding mask

### ISSUE THREAD

**Issue by m96-chan** [m96-chan/web-xpu-ops#77]
`attention`, `flash_attention` and `gqa` can mask one way: `causal`, positioned by `queryOffset`. There is no way to say *"key `j` is padding, never attend to it"*.

That is fine for a decoder, where every key is real and causality is the only rule. It does not cover an encoder.

## What needs it

The RF-DiT in [m96-chan/voxshot#115](https://github.com/m96-chan/voxshot/issues/115) passes an explicit key mask to **every one of its four** attention call sites (`irodori_tts/model.py`):

```python
attn_mask = key_mask[:, None, None, :]
y = F.scaled_dot_product_attention(q, k, v, attn_mask=attn_mask, ...)
```

Its joint attention concatenates text, speaker and caption context into one key sequence (`torch.cat(context_masks, dim=1)`) — three conditioning streams of different, per-sample lengths in one batched dispatch. Without a key mask, padding positions contribute to the softmax and every conditioned output is quietly wrong. Not NaN, not a crash: slightly wrong prosody.

This is not one model's quirk. Any batched encoder over variable-length sequences needs it, and this library currently has no answer for encoders at all.

## Honest scope note

At batch 1 — which is the browser case, and voxshot's case — padding can be avoided entirely by slicing each sequence to its true length before the dispatch. So this is not a blocker for that integration; it is the difference between an op set that serves decoders and one that serves encoders too. Worth doing on its own terms, not urgently.

## Conventions

`torch.nn.functional.scaled_dot_product_attention`'s `attn_mask`, which accepts two types and means different things by them:

- **bool**: `True` means *attend*, `False` means *masked*. This trips people constantly — it is the opposite polarity of `key_padding_mask` in `nn.MultiheadAttention`, where `True` means *ignore*. Pick one, name which torch API it matches, and say so in `reference.ts`.
- **float**: **added** to the scores before softmax, so `-inf` masks and `0` does not. This is the form that composes with ALiBi, which this library already has as a separate op producing exactly such a bias.

Given `alibi` already exists and its README entry says "masking is the caller's", the additive float form is the one that makes the two compose. A boolean convenience on top is a separate decision.

Other things that need deciding rather than defaulting:

- **Broadcasting.** Torch broadcasts `attn_mask` against `[B, H, L, S]`, so `[B, 1, 1, S]` (a key-padding mask) and `[B, H, L, S]` (a full bias) are both legal. Supporting only the first is a defensible scope cut; supporting neither is not. Say which.
- **Combining with `causal`.** Torch raises if `is_causal` and `attn_mask` are both given. Following that is cheaper than defining an intersection nobody asked for.
- **A fully masked row.** `ops/attention/reference.ts:35-39` states the current contract: `queryOffset >= 0` guarantees every row sees key 0, so no row is ever fully masked, and the kernels deliberately do not guard for it — the reference would give NaN where torch gives zeros. **A key mask breaks that guarantee.** A sample whose mask is all-`False` is now reachable, and the existing comment stops being true the moment this lands. Either guard, or restate the contract as the caller's — but the current text cannot survive unchanged.
- **`flash_attention` and `gqa` too**, or the mask silently decides which kernel a caller may use. If they are staged, say which come later.

## Done when

- [ ] The mask type (additive float, boolean, or both) is decided, with the torch API it matches named in `reference.ts`
- [ ] The reference matches `scaled_dot_product_attention` with a `[B, 1, 1, S]` key mask, checked numerically
- [ ] The fully-masked-row contract at `ops/attention/reference.ts:35-39` is corrected — a NaN row is a wrong answer that looks like a result
- [ ] `attention` masks and matches; `flash_attention` and `gqa` either match or have an issue saying when
- [ ] The interaction with `causal` is defined (raise, most likely)
- [ ] README op table and CHANGELOG updated
